LXQt-Desktop: Missing profiles

Besanon opened the pull request:
start for wayland session with labwc, 2 more profiles

Commit:
Create abstraction for lxqt desktop group

first file for the LXQT 2.0 desktop group

Commit:
Update lxqt

Commit:
xdg-desktop abstraction added

Commit:
removing tabs

Commit:
Create startlxqt

starter file for LXQT Desktop

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Create startlxqt

Commit:
fixing startlxqt

I use sddm as display manager

I cant remove the other file - only use graphical env., sorry

After startlxqt i would add  2 lines to sddm to enable the start of LXQT desktop

Commit:
Delete apparmor.d/profiles-s-z/startlxqt

Commit:
indented by 2 spaces (like other entries)

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Update sddm

Enable sddm to start an lxqt desktop session

Commit:
Create lxqt-session

lxqt-session to be started by startlxqt. Display manager: sddm

Commit:
Update lxqt-session

Commit:
Update lxqt-session

Commit:
removed trailing whitespace

Commit:
Update kscreen_backend_launcher to support lxqt desktop

is needed for several complaints:
DENIED  kscreen_backend_launcher open owner @{user_config_dirs}/lxqt/lxqt.conf comm=kscreen_backend requested_mask=r denied_mask=r
DENIED  kscreen_backend_launcher open /usr/share/lxqt/lxqt.conf comm=kscreen_backend requested_mask=r denied_mask=r
DENIED  kscreen_backend_launcher open owner @{user_config_dirs}/lxqt/session.conf comm=kscreen_backend requested_mask=r denied_mask=r
DENIED  kscreen_backend_launcher open /usr/share/lxqt/session.conf comm=kscreen_backend requested_mask=r denied_mask=r

Commit:
Update lxqt-session

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Create lxqt-panel

Commit:
Update lxqt-panel

Commit:
Update lxqt-panel

Commit:
Update lxqt-panel

Commit:
Merge branch 'roddhjav:main' into main

Commit:
fix conflicting x

Commit:
Update lxqt-panel

add child-open

Commit:
remove  include <abstractions/app-launcher-user>

you think its too permissive to have app-launcher-user here, right?

Commit:
Update lxqt-panel

add needed programs

Commit:
Update lxqt-panel

turning back to layout of corresponding xfce file.

Commit:
Create lxqt-globalkeysd

Commit:
Create lxqt-about

Commit:
Create lxqt-leave

Commit:
Create lxqt-runner

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Update lxqt-leave

Commit:
Update lxqt-runner

Commit:
Update lxqt-globalkeysd

Commit:
remove video in lxqt-about

Commit:
Update lxqt-about

Commit:
Update lxqt-runner

Commit:
remove abstr. in lxqt-globalkeysd

Commit:
remove abstr. in lxqt-runner

Commit:
remove abstr. in lxqt-leave

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Create lxqt-config-notificationd

Commit:
Create lxqt-config-locale

Commit:
Create lxqt-config-printer

Commit:
Create lxqt-config-file-associations

Commit:
Create lxqt-config-powermanagement

Commit:
enable wayland-session for lxqt 2.1

startlxqtwayland for starting the session, support for labwc and kwin_wayland

Commit:
Update lxqt-config-printer

Commit:
Update lxqt-config-powermanagement

Commit:
Update sddm

Commit:
Update sddm

Commit:
adapt pci-rules

ok, havent seen this profile yet. I will change that in lxqt-powermanagement as well and check the other profiles

Commit:
Update lxqt-config-powermanagement

Commit:
Update lxqt-config-powermanagement

Commit:
Update lxqt-config-powermanagement

Commit:
Update lxqt-config-powermanagement

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Create startlxqtwayland

basic support for labwc

Commit:
Create lxqt-notificationd

Commit:
Update startlxqtwayland

Commit:
Create lxqt-config-brightness

Commit:
Update startlxqtwayland

Commit:
Update startlxqtwayland

Commit:
Update lxqt-config-brightness

Commit:
Update lxqt-config-notificationd

Commit:
Update lxqt-config-notificationd

Commit:
Update lxqt-config-notificationd

Commit:
Update lxqt-notificationd

Commit:
Update lxqt-notificationd

Commit:
Update labwc for use with lxqt

Commit:
Merge branch 'roddhjav:main' into main

Commit:
Create ControlPanel

Commit:
Update lxqt-about

Commit:
Create lxqt-backlight_backend

Commit:
Create lxqt-config

Commit:
Create lxqt-config-appearance

Commit:
Create lxqt-config-globalkeyshortcuts

Commit:
Create lxqt-config-input

Commit:
Create lxqt-config-monitor

Commit:
Create lxqt-config-session

Commit:
Create lxqt-policykit-agent

Commit:
Update lxqt-config-globalkeyshortcuts

Commit:
Update lxqt-config-session

Commit:
Update lxqt-about

Commit:
Update startlxqtwayland

Commit:
Update lxqt-policykit-agent

Commit:
Update lxqt-config-input

Commit:
Update lxqt-config-appearance

Commit:
Update lxqt-config

Commit:
Update lxqt-config-session

Commit:
Update lxqt-backlight_backend

Commit:
Update lxqt-config-input

Commit:
Update lxqt-policykit-agent

Commit:
Update lxqt-config-input

Commit:
Update lxqt-session

Commit:
Update lxqt-backlight_backend

Commit:
Update startlxqtwayland

Commit:
Update startlxqtwayland

Commit:
Update lxqt-config-input

Commit:
Update lxqt-config-appearance

Commit:
Update lxqt-config-appearance

Commit:
Update ControlPanel

Commit:
Update lxqt-config-session

Commit:
Update lxqt-config-input

Commit:
Update startlxqtwayland

Commit:
Update lxqt-config-input

Commit:
Update lxqt-config-monitor

Commit:
Update lxqt-config-session

Commit:
Update startlxqtwayland

Commit:
Update lxqt-config

Commit:
Update ControlPanel

Commit:
Update lxqt-config-session

Commit:
Update lxqt-config

roddhjav commented on apparmor.d/groups/lxqt/ControlPanel:
```diff
@@ -0,0 +1,38 @@
+
+  owner /tmp/@{int} r,
+
+  /dev/pts/@{int} rw,
+  /dev/tty rw,
```
Use the consoles abs instead.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config:
```diff
@@ -0,0 +1,67 @@
+  @{bin}/nm-connection-editor rPx,
+  @{bin}/ControlPanel rPx,
+  @{bin}/qt6ct rix,
+  @{bin}/xdg-open rPx,
```
This needs to be handled as:
```sh
  @{open_path}                                   rPx -> child-open-any,
```
See: https://apparmor.pujol.io/development/internal/#open-resources

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config:
```diff
@@ -0,0 +1,67 @@
+  owner @{user_config_dirs}/lxqt/ r,
+  owner @{user_config_dirs}/lxqt/#@{int} rw,
+  owner @{user_config_dirs}/lxqt/lxqt-config.conf.lock rwk,
+  owner @{user_config_dirs}/lxqt/lxqt-config.conf.@{rand6} rwl -> @{user_config_dirs}/lxqt/#@{int8}7,
+  owner @{user_config_dirs}/lxqt/lxqt-config.conf.@{rand6} rwl -> @{user_config_dirs}/lxqt/#@{int8}2,
```
To merge the rules, you should use `-> @{user_config_dirs}/lxqt/#@{int},` (beware aa-log output can still be simplified).

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-input:
```diff
@@ -0,0 +1,104 @@
+
+  @{sys}/bus/**/devices/ r,   # ALL under /sys/bus/* is asked for read
+  @{sys}/class/**/ r, # ALL but usbmisc under /sys/class is being read
+  @{sys}/devices/**/uevent r,
+  @{sys}/devices/platform/**/uevent r,
+  @{sys}/devices/platform/cpu/**/uevent r,
+  @{sys}/devices/system/machinecheck/**/uevent r,
+  @{sys}/devices/pnp@{int}/**/uevent r,
+  @{sys}/devices/system/clockevents/clockevent@{int}/uevent r,
+  @{sys}/devices/system/cpu/cpu@{int}/uevent r,
+  @{sys}/devices/system/memory/memory@{int}/uevent r,
+  @{sys}/devices/virtual/devlink/**/uevent r,
+  @{sys}/devices/virtual/mem/**/uevent r,
+  @{sys}/devices/virtual/bdi/@{int}:@{int}/uevent r,
+  @{sys}/devices/virtual/block/loop@{int}/uevent r,
+  @{sys}/devices/virtual/input/**/uevent r,
+  @{sys}/devices/virtual/memory_tiering/memory_tier@{int}/uevent r,
+  @{sys}/devices/virtual/misc/**/uevent r,
+  @{sys}/devices/virtual/sound/seq/uevent r,
+  @{sys}/devices/virtual/sound/timer/uevent r,
+  @{sys}/devices/virtual/sound/ctl-led/uevent r,
+  @{sys}/devices/virtual/thermal/thermal_zone@{int}/uevent r,
+  @{sys}/devices/virtual/thermal/cooling_device@{int}/uevent r,
+  @{sys}/devices/virtual/tty/**/uevent r,
+  @{sys}/devices/virtual/vc/vcsu@{int}/uevent r,
+  @{sys}/devices/virtual/vc/vcsa@{int}/uevent r,
+  @{sys}/devices/virtual/vc/vcs@{int}/uevent r,
+  @{sys}/devices/LNXSYSTM:00/PNP*/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:@{rand2}/LNXTHERM:@{rand2}/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:00/PNP*/PNP*/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:00/HPIC*/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:@{rand2}/PNP*/device*/device*/device*/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:@{rand2}/PNP*/device*/device*/device*/device*/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:@{rand2}/AMDI*/**/wakeup@{int}/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:@{rand2}/PNP*/uevent r,
+  @{sys}/devices/LNXSYSTM:00/LNXSYBUS:@{rand2}/PNP*/**/wakeup/wakeup@{int}/uevent r,
```
The first rule include all the other ones. So only keep the first one.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-monitor:
```diff
@@ -0,0 +1,42 @@
+
+  @{exec_path} mr,
+
+  /var/cache/fontconfig/{,**} rw,
```
Use  `<abstractions/fontconfig-cache-write>` instead.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-session:
```diff
@@ -0,0 +1,58 @@
+  owner @{user_config_dirs}/QtProject.conf rw,
+  owner @{user_config_dirs}/QtProject.conf.@{rand6} rwkl,
+  owner @{user_config_dirs}/QtProject.conf.lock rwk,
+  owner @{user_config_dirs}/autostart/*.desktop rw,
+  owner @{user_config_dirs}/autostart/lxqt-config-monitor-autostart.desktop  r,
```
You can remove the second rule, as the first one include it already.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-notificationd:
```diff
@@ -0,0 +1,55 @@
+
+  owner /tmp/@{int} r,
+
+  /dev/tty rw,
+  /dev/tty@{int} rw,
+  owner /dev/tty@{int} rw,
```
Use the consoles abs instead

roddhjav commented on apparmor.d/groups/lxqt/lxqt-policykit-agent:
```diff
@@ -0,0 +1,55 @@
+
+  @{lib}/polkit-[0-9]/polkit-agent-helper-[0-9] rPx,
+
+  /usr/share/lxqt/translations/lxqt-policykit-agent/lxqt-policykit-agent_de.qm r,
```
That should be already in the lxqt abs.

Commit:
Update ControlPanel

Commit:
Update lxqt-backlight_backend

Besanon commented on apparmor.d/groups/lxqt/lxqt-config-appearance:
```diff
@@ -0,0 +1,51 @@
+  @{exec_path} mr,
+
+  @{bin}/gsettings rPx,
+  @{bin}/pcmanfm-qt rPx,
```
ah, i missed that. There are some additional profiles. Add it to the collection

Commit:
Update lxqt-config

Commit:
Create pcmanfm-qt

Commit:
Update lxqt-config-brightness

Commit:
Update lxqt-config-input

Commit:
Update lxqt-config-monitor

Commit:
Update lxqt-config-monitor

Commit:
Update lxqt-config-session

Commit:
Update lxqt-notificationd

Commit:
Update lxqt-policykit-agent

Commit:
Update lxqt

Commit:
Update lxqt-config-input

Commit:
Update lxqt-config-brightness

Commit:
Update lxqt-config-notificationd

Commit:
Update lxqt-config-monitor

Commit:
Update pcmanfm-qt

Commit:
Update pcmanfm-qt

Commit:
Update lxqt-notificationd

Commit:
Update lxqt-config-input

Commit:
Update lxqt

Commit:
Update lxqt-session

Commit:
Update lxqt-panel

Commit:
Update programs

Commit:
Create qterminal

roddhjav commented on apparmor.d/groups/lxqt/pcmanfm-qt:
```diff
@@ -0,0 +1,110 @@
+  owner /usr/**   rw,
+        /var/     r,
+        /var/**   r,
+  owner /var/**   rw,
```
No, the policy in this project is to only allow user files. This can be changed by the end user, see https://apparmor.pujol.io/configuration/#file-browsers.

Here you need:
https://github.com/roddhjav/apparmor.d/blob/eda29668ae75d8b42412f35e3737230c6a626c09/apparmor.d/groups/gnome/nautilus#L84-L105

End these rules need to be moved before `owner @{user_cache_dirs}...`

roddhjav commented on apparmor.d/groups/lxqt/pcmanfm-qt:
```diff
@@ -0,0 +1,110 @@
+  @{lib}/menu-cache/menu-cached rPx,
+  @{lib}exec/menu-cache/menu-cache-gen rix,
+
+  owner @{user_cache_dirs}/pcmanfm-qt/**  r,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/ r,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/recent-files.conf.lock rwk,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/desktop-items-0.conf.@{rand6} l -> @{user_config_dirs}/pcmanfm-qt/lxqt/#@{int},
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/dir-settings.conf~ l -> @{user_config_dirs}/pcmanfm-qt/lxqt/dir-settings.conf,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/desktop-items-eDP-@{int}.conf.lock rwk,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/settings.conf.lock rwk,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/settings.conf.@{rand6} l -> @{user_config_dirs}/pcmanfm-qt/lxqt/#@{int},
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/desktop-items-0.conf.lock rwk,
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/desktop-items-eDP-@{int}.conf l -> @{user_config_dirs}/pcmanfm-qt/lxqt/#@{int},
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/recent-files.conf.@{rand6} l -> @{user_config_dirs}/pcmanfm-qt/lxqt/#@{int},
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/settings.conf.lock.* rwk,
```
As `pcmanfm-qt` *own* its own config folder, you should simplify this as:

```
  owner @{user_cache_dirs}/pcmanfm-qt/{,**} rw,
  owner @{user_config_dirs}/pcmanfm-qt/ rw,
  owner @{user_config_dirs}/pcmanfm-qt/** rwlk ->  @{user_config_dirs}/pcmanfm-qt/**,
```

roddhjav commented on apparmor.d/groups/lxqt/pcmanfm-qt:
```diff
@@ -0,0 +1,110 @@
+
+  network netlink raw,
+
+  #aa:dbus own bus=session name=org.pcmanfm.PCManFM
```
Where is the `org.pcmanfm.PCManFM` dbus interface defined? I did not find any

roddhjav replied:
My bad, it is needed:
https://github.com/lxqt/pcmanfm-qt/blob/66ca7b2c7e3ca9281f5ee41c7159485d8d14f6fe/pcmanfm/application.cpp#L103-L105

You can also add `#aa:dbus own bus=session name=org.freedesktop.FileManager1`:
https://github.com/lxqt/pcmanfm-qt/blob/66ca7b2c7e3ca9281f5ee41c7159485d8d14f6fe/pcmanfm/application.cpp#L134-L136

Besanon replied:
i took it from the output of D-spy - journalctl didnt complain about that. But if its not sure we should dprop that line

roddhjav replied:
We cannot drop it, it is needed.

> journalctl didnt complain about that

That's because you are on arch, dbus mediation is not enabled. If you do your tests on ubuntu, you will discover a whole new world of deny.

roddhjav commented on apparmor.d/groups/lxqt/pcmanfm-qt:
```diff
@@ -0,0 +1,110 @@
+  include <abstractions/bus/org.freedesktop.UDisks2>
+  include <abstractions/deny-sensitive-home>
+  include <abstractions/devices-usb>
+  include <abstractions/disks-read>
```
Again, a file manager should not need this.

roddhjav commented on apparmor.d/groups/lxqt/pcmanfm-qt:
```diff
@@ -0,0 +1,110 @@
+  network netlink raw,
+
+  #aa:dbus own bus=session name=org.pcmanfm.PCManFM
+  #aa:exec kioworker
```
This is not a dbus rule, move it with the other transitions.

roddhjav commented on apparmor.d/groups/lxqt/pcmanfm-qt:
```diff
@@ -0,0 +1,110 @@
+  @{exec_path} mr,
+
+  @{lib}/menu-cache/menu-cached rPx,
+  @{lib}exec/menu-cache/menu-cache-gen rix,
```
* There is no profile (yet) for `menu-cached`
* `@{lib}exec` -> `@{lib}`

Besanon commented:
Hello Alex,
i havent reflected all your modifications in the profile kwin_wayland. So, when using the changed version i get 
ENIED  kwin_wayland open owner /var/lib/sddm/.config/kwinoutputconfig.json comm=kwin_wayland requested_mask=r denied_mask=r
DENIED  kwin_wayland open owner /var/lib/sddm/.config/kwinoutputconfig.json comm=kwin_wayland requested_mask=wc denied_mask=wc
DENIED  kwin_wayland open /usr/share/kwin-wayland/effects/ comm="Thread (pooled)" requested_mask=r denied_mask=r
DENIED  kwin_wayland open /usr/share/kwin-wayland/scripts/ comm=kwin_wayland requested_mask=r denied_mask=r
DENIED  kwin_wayland open owner @{user_config_dirs}/lxqt/lxqt.conf comm=kwin_wayland requested_mask=r denied_mask=r
DENIED  kwin_wayland open /usr/share/lxqt/lxqt.conf comm=kwin_wayland requested_mask=r denied_mask=r
DENIED  kwin_wayland open owner @{user_config_dirs}/lxqt/session.conf comm=kwin_wayland requested_mask=r denied_mask=r
DENIED  kwin_wayland open /usr/share/lxqt/session.conf comm=kwin_wayland requested_mask=r denied_mask=r
DENIED  kwin_wayland open /usr/share/kwin-wayland/effects/ comm="Thread (pooled)" requested_mask=r denied_mask=r
DENIED  kwin_wayland open /usr/share/kwin-wayland/scripts/ 

What should we do about that? We could add abstractions/lxqt  to the profile - i will check that

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-brightness:
```diff
@@ -22,11 +23,39 @@ profile lxqt-config-brightness @{exec_path} {
 
   /dev/tty rw,
 
+  profile pkexec {
+    include <abstractions/base>
+    include <abstractions/app/pkexec>
+    include <abstractions/python>
+
+    ptrace read peer=aa-notify,
+
+    @{sbin}/apparmor_parser Px,
+    @{lib}/@{python_name}/site-packages/apparmor/update_profile.py ix,
+
+    /usr/share/apparmor/** r,
+    /usr/share/terminfo/** r,
+
+    @{etc_ro}/inputrc r,
+    @{etc_ro}/inputrc.keys r,
+
+    /etc/apparmor.d/ r,
+    /etc/apparmor.d/** rw,
+    /etc/apparmor/* r,
+
+    @{PROC}/@{pid}/mounts r,
+
+    include if exists <local/lxqt-config-brightness_pkexec>
+  }
```
No, I did not ask you to fully copy paste the pkexec subprofile from aa-notify.

By design, this subprofile only allow editing and reloading an apparmor profile. Here you need to only allow the things `lxqt-config-brightness` needs to do with privileged right.

roddhjav commented on apparmor.d/groups/lxqt/qterminal:
```diff
@@ -0,0 +1,72 @@
+
+  owner @{user_cache_dirs}/icon-cache.kcache rw,
+  owner @{user_config_dirs}/lxqt/lxqt.conf r,
+  owner @{user_config_dirs}/qterminal.org/** rw,
```
Same as for pcmanfm. This is qterminal own directory, allow everything in it.

roddhjav commented on apparmor.d/groups/lxqt/qterminal:
```diff
@@ -0,0 +1,72 @@
+
+  owner @{user_cache_dirs}/icon-cache.kcache rw,
+  owner @{user_config_dirs}/lxqt/lxqt.conf r,
+  owner @{user_config_dirs}/qterminal.org/** rw,
+  owner @{user_config_dirs}/qterminal.org/#@{int} rwk,
+  owner @{user_config_dirs}/qterminal.org/qterminal.ini.lock rwk,
+  owner @{user_config_dirs}/qterminal.org/qterminal.ini.@{rand6} rwk,
+  owner @{user_config_dirs}/qterminal.org/qterminal.ini.@{rand6}  l -> @{user_config_dirs}/qterminal.org/#@{int},
```
Same as for pcmanfm. This is qterminal own directory, allow everything in it.

Commit:
Update pcmanfm-qt

Commit:
Update lxqt-config-notificationd

Commit:
Update lxqt-panel

Commit:
Update qterminal

Commit:
Update pcmanfm-qt

Commit:
Update lxqt-notificationd

Commit:
Update lxqt-config-notificationd

Commit:
Update lxqt-config-brightness

So far that seems to work without complains. sys-statements required both in subprofile and profile

Commit:
Update kwin_wayland

These 4 modifications are needed to read lxqt configs
abstractions/lxqt contains a bit more, i dont think that it is necessary

Commit:
Update kreadconfig

reflect name changes

Commit:
Update kbuildsycoca

Commit:
Update kwin_wayland

after enabling the ksycoca6

Commit:
Update xdg-mime

Commit:
Update xdg-settings

Commit:
Update xdg-mime

Commit:
Update kbuildsycoca

Commit:
Update kglobalacceld

reflecting newer version

Commit:
Update pcmanfm-qt

some more statements missing

Commit:
Update kwin_wayland

Besanon commented:
Hello Alex, i have resolved some differnces in the sddm-profiles, however i am asked to include
  @{bin}/xargs                rix,  in sddm   - and -
as long as i put the 3 sddm-profiles in complain mode there is no issue with the sddm.service. As soon as i enforce them
(even just a single one) i am getting:
ug 17 09:37:31 host sddm-helper[2782]: [PAM] Preparing to converse...
Aug 17 09:37:31 host sddm-helper[2782]: [PAM] Conversation with 1 messages
Aug 17 09:37:32 host sddm-helper[2782]: [PAM] returning.
Aug 17 09:37:32 host sddm[1546]: Authentication for user  "user"  successful
Aug 17 09:37:32 host sddm-helper[2782]: pam_unix(sddm:session): session opened for user user(uid=1000) by user(uid=0)
Aug 17 09:37:32 host sddm[1546]: Auth: sddm-helper exited successfully
Aug 17 09:37:32 host sddm[1546]: Greeter stopped. SDDM::Auth::HELPER_SUCCESS
Aug 17 09:37:33 host sddm-helper[2782]: pam_systemd(sddm:session): Failed to set ambient capabilities, ignoring: Die Operation ist nicht erlaubt
Aug 17 09:37:33 host sddm-helper[2782]: Starting Wayland user session: "/usr/share/sddm/scripts/wayland-session" "startlxqtwayland"
Aug 17 09:37:33 host sddm[1546]: Session started true

on systemctl status sddm
[

Commit:
Create xdg-desktop-portal-lxqt

Commit:
Update xdg-desktop-portal-lxqt

Commit:
Update kwin_wayland

sddm hangs with curser shown,
starts without complain when adding this line

pam error still there, maybe an arch related issue

Besanon commented:
Anything left for me to check?

Commit:
Update kwin_wayland

sorry for the german-only assignment

roddhjav committed:
Merge branch 'main' into main

Commit:
Update pcmanfm-qt

add the dbus line again!
#aa:dbus own bus=session name=org.pcmanfm.PCManFM

Commit:
Update pcmanfm-qt

Commit:
Update xdg-desktop-portal-lxqt

ah this was the issue

Commit:
Update pcmanfm-qt

Besanon commented:
Hello Alex,
is the profile su the cause of the error? 
Sep 07 20:50:47 pkrvm7jw40e0xgp apparmor.systemd[13490]: AppArmor parser error for /etc/apparmor.d in profile /etc/apparmor.d/su at line 39: attach_disconnected_path value must begin with a /

roddhjav commented:
Yes, this one is because of me don't worry.

Commit:
reload after su profile alteration

Commit:
Create lxqt-admin-time

Commit:
Create lxqt-admin-user

Commit:
Create lxqt-admin-user-helper

Commit:
Create featherpad

Commit:
Create lximage-qt

Commit:
Create lxqt-archiver

Commit:
Update lxqt-admin-time

Commit:
Create lxqt-openssh-askpass

Commit:
Create lxqt-powermanagement

Commit:
Update lximage-qt

Commit:
Update lxqt-powermanagement

Commit:
Update featherpad

Commit:
Update lximage-qt

Commit:
Update lxqt-powermanagement

Commit:
Update featherpad

Commit:
Update lxqt-admin-user-helper

Commit:
Update lxqt-archiver

Commit:
Update lxqt-openssh-askpass

Commit:
Update lxqt-admin-user

Commit:
Update lxqt-admin-user-helper

Commit:
Update lximage-qt

Commit:
Update featherpad

Commit:
Update lxqt-admin-user-helper

Commit:
Update lxqt-admin-time

Commit:
Update lxqt-admin-time

Commit:
Update lxqt-admin-user

Commit:
Update lxqt-openssh-askpass

Commit:
Update lxqt-archiver

Commit:
Update featherpad

Commit:
Update lxqt-admin-user

Commit:
Update lxqt-admin-time

Commit:
Update featherpad

Commit:
Update lxqt-admin-time

Commit:
Update lxqt-config-input

Commit:
Update featherpad

Commit:
Merge branch 'main' into main

Commit:
Update pcmanfm-qt

Commit:
Update lxqt-config-input

Commit:
Update featherpad

Commit:
Merge branch 'main' into main

Commit:
Merge branch 'main' into main

Besanon commented:
I updated my repository for LXQt where the most recent changes and some additional profiles can be found:

https://github.com/Besanon/apparmor-LXQt

roddhjav commented:
I am going to close this PR. Not because of your work, but because merging this would require too much testing and maintenance time.

This is not a matter of simply adding tests for the LXQt desktop environment; it is a matter of supporting the environment today, and tomorrow, on all distributions. I simply do not have the time and the resources for this.
And I need to prioritize tasks toward the general stability and security of the full project first.

Besanon commented:
I understand your decision. 
The profiles of Gnome, KDE and XFCE Desktops are too much for a continuous development if you consider all major distributions.
In KDE i see so much small changes in the profiles (those which a relevant for LXQT/KWIN). There are still some to be reflected in your collection of profiles. If you like i will put them one by one in single pull requests.
Regarding XFCE there are a whole bunch of programs (all the settings-related and more) still without a profile AND a lot of unresolved references to the glycin namespaces. Some of them now have a shiftmade profile. As soon as V5 of apparmor sheds more light on the handling of these glycin related requests I could show you the results.
I know LXQt is a minor DE with low acceptance. Nevertheless i will continue to adapt these profiles to let them run along with your project. Maybe you could put a hint in your lxqt group that there is more (2/3) to find in my repository. Its indeed a very small memory footprint (790MB after login running with KWIN/Wayland) and its running very well in X and wayland sessions
I do not use user-namespaces, dropping the support in sysctl. Bubblewrap cant work properly., yes. The developers in the gnome project are not going to resolve the cascaded confinements of glycin and apparmor as valoq complained in vain.  Dont know if these cascading sandboxes are of real use - at last they are all software.
Good luck
